bin/init.sh

```
@@ -61,104 +61,143 @@ export ISTIO_OUT=${ISTIO_OUT:-${ISTIO_BIN}}
   export PROXY_REPO_SHA
 fi
 
-# Normally set by the Makefile.
-ISTIO_ENVOY_VERSION=${ISTIO_ENVOY_VERSION:-${PROXY_REPO_SHA}}
-ISTIO_ENVOY_DEBUG_URL=${ISTIO_ENVOY_DEBUG_URL:-https://storage.googleapis.com/istio-build/proxy/envoy-debug-${ISTIO_ENVOY_VERSION}.tar.gz}
-ISTIO_ENVOY_RELEASE_URL=${ISTIO_ENVOY_RELEASE_URL:-https://storage.googleapis.com/istio-build/proxy/envoy-alpha-${ISTIO_ENVOY_VERSION}.tar.gz}
-# TODO Change url when official envoy release for MAC is available
-ISTIO_ENVOY_MAC_RELEASE_URL=${ISTIO_ENVOY_MAC_RELEASE_URL:-https://github.com/istio/proxy/releases/download/1.0.2/istio-proxy-1.0.2-macos.tar.gz}
-
-# Normally set by the Makefile.
-# Variables for the extracted debug/release Envoy artifacts.
-ISTIO_ENVOY_DEBUG_DIR=${ISTIO_ENVOY_DEBUG_DIR:-"${OUT_DIR}/${GOOS}_${GOARCH}/debug"}
-ISTIO_ENVOY_DEBUG_NAME=${ISTIO_ENVOY_DEBUG_NAME:-"envoy-debug-$ISTIO_ENVOY_VERSION"}
-ISTIO_ENVOY_DEBUG_PATH=${ISTIO_ENVOY_DEBUG_PATH:-"$ISTIO_ENVOY_DEBUG_DIR/$ISTIO_ENVOY_DEBUG_NAME"}
-ISTIO_ENVOY_RELEASE_DIR=${ISTIO_ENVOY_RELEASE_DIR:-"${OUT_DIR}/${GOOS}_${GOARCH}/release"}
-ISTIO_ENVOY_RELEASE_NAME=${ISTIO_ENVOY_RELEASE_NAME:-"envoy-$ISTIO_ENVOY_VERSION"}
-ISTIO_ENVOY_RELEASE_PATH=${ISTIO_ENVOY_RELEASE_PATH:-"$ISTIO_ENVOY_RELEASE_DIR/$ISTIO_ENVOY_RELEASE_NAME"}
+# Allow user-override for a local Envoy build.
+USE_LOCAL_PROXY=${USE_LOCAL_PROXY:-0}
+ISTIO_ENVOY_LOCAL=${ISTIO_ENVOY_LOCAL:-$(realpath "${ISTIO_GO}/../proxy/bazel-bin/src/envoy/envoy")}
+if [[ ${USE_LOCAL_PROXY} == 1 ]] ; then
+  echo "Using istio-proxy image from local workspace: ${ISTIO_ENVOY_LOCAL}"
+  if [[ ! -f "${ISTIO_ENVOY_LOCAL}" ]]; then
+    echo "Error: missing istio-proxy from local workspace: ${ISTIO_ENVOY_LOCAL}. Check your build path."
+    exit 1
+  fi
+
+  # Point the native paths to the local envoy build.
+  if [[ "$LOCAL_OS" == "Darwin" ]]; then
+    ISTIO_ENVOY_MACOS_RELEASE_PATH=${ISTIO_ENVOY_LOCAL}
```

This might be intentional (but then, we need a warning, since it's going to fail for a weird reason), but calling `make docker.proxyv2` on macOS with `USE_LOCAL_PROXY=1` is going to fail, since `ISTIO_ENVOY_LINUX_RELEASE_DIR` is going to be unset.

We should either error out here saying that this is unsupported combination, or silently fallback to using non-local Linux binaries and produce working Docker builds.

---

Good point. I think at this point the best thing to do is to revert the logic so that we always set these variables and then optionally override with a local build.